### PR TITLE
GNU Solfege 3.22.2 (new formula)

### PR DESCRIPTION
**Pull request by ram-on**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Added GNU Solfege 3.22.2 formula.

### CODE REVIEW

**Comment by ram-on** [comment]
>     * Formulae should not have optional or recommended dependencies
> 
>     * Formulae should not have optional or recommended dependencies
> 
>     * Formulae should not have optional or recommended dependencies

Hi @fxcoudert. Is the above a must have?  (Since the Formula Cookbook does not discourage the usage of `:optional` and `:recommended`...)

**Comment by ram-on** [comment]
@SMillerDev timidity was added as`:recommended` since it can be replaced by mpg123 or vorbis-tools (hence the `:optional` "tag")

**Comment by ram-on** [comment]
Can anyone please merge this?

File: Formula/solfege.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  homepage "https://www.gnu.org/software/solfege/"
+  url "https://ftp.gnu.org/gnu/solfege/solfege-3.22.2.tar.bz2"
+  mirror "https://ftpmirror.gnu.org/solfege/solfege-3.22.2.tar.bz2"
+  sha256 "5a6fcd46b0ce33817752f5b04e053b5342d8d2e3c8b3df2ea789cf3366a38b10"
+
+  depends_on "gettext"      => :build
+  depends_on "pkg-config"   => :build
+  depends_on "librsvg"
+  depends_on "pygtk"
+  depends_on "timidity"     => :recommended
```
Comment (fxcoudert) [inline Formula/solfege.rb]: Make `timidity` mandatory, remove the options below.

File: Formula/solfege.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "mpg123"       => :optional
+  depends_on "vorbis-tools" => :optional
+
+  def install
+    inreplace "Makefile.in", "cp --parents", "rsync -R"
```
Comment (fxcoudert) [inline Formula/solfege.rb]: Why?

File: Formula/solfege.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    inreplace "default.config", "/usr/bin/", "#{HOMEBREW_PREFIX}/bin/"
```
Comment (fxcoudert) [inline Formula/solfege.rb]: Shouldn't this be done already by running `configure` with `prefix`? If not, report it as a bug.

File: Formula/solfege.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    inreplace "solfege/mainwin.py", "webbrowser.open(filename)", "webbrowser.open(\"file://\" + filename)"
+
+    system "./configure", "--prefix=#{prefix}", "--disable-oss-sound"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/solfege", "--version"
```
Comment (fxcoudert) [inline Formula/solfege.rb]: We need a test that exercises the some of the functionality of the app.

File: Formula/solfege.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "librsvg"
+  depends_on "pygtk"
+  depends_on "timidity"     => :recommended
+  depends_on "mpg123"       => :optional
+  depends_on "vorbis-tools" => :optional
+
+  def install
+    inreplace "Makefile.in", "cp --parents", "rsync -R"
+    inreplace "default.config", "/usr/bin/", "#{HOMEBREW_PREFIX}/bin/"
+    inreplace "solfege/mainwin.py", "webbrowser.open(filename)", "webbrowser.open(\"file://\" + filename)"
```
Comment (fxcoudert) [inline Formula/solfege.rb]: Was this bug reported?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Pull request closed without merging**